docs/ref/models/querysets.txt

```diff
@@ -2516,6 +2516,23 @@ them, but it has a few caveats:
 * If updating a large number of columns in a large number of rows, the SQL
   generated can be very large. Avoid this by specifying a suitable
   ``batch_size``.
+* When updating a large number of objects, be aware that ``bulk_update()``
+  prepares all of the ``WHEN`` clauses for every object across all batches
+  before executing any queries. This can require more memory than expected. To
+  reduce memory usage, you can use an approach like this::
+
+    from itertools import islice
+
+    batch_size = 100
+    entry_iter = (Entry.objects.get(pk=i) for i in range(1000))
+    while True:
+        batch = list(islice(entry_iter, batch_size))
+        if not batch:
+            break
```

Comment:
I do understand that it's a succinct way of getting model instances for the stake of the example but this slightly bugs me as this means we'll perform 1000 queries 😅 

What about 
```suggestion
    ids_iter = range(1000)
    while ids := list(islice(ids_iter, batch_size)):
        batch = Entry.objects.filter(ids__in=ids)
```